### ISSUE TITLE
can we use "External Data" and "Custom Row Format" together  

### ISSUE THREAD

**Issue by jitchavan** [GriddleGriddle/Griddle#348]
Hi,

i trying to use **"External Data**" and "**Custom Row Format**"  together but with no success yet.
so i m trying to "Custom Row Format"  only and use state to update results from external source.

any thoughts on this.

Thanks.

**Comment by ryanlanciaux** [comment]
Sorry that's not working out for you! Could you please post how you're defining your Griddle component? Thanks!

**Comment by jitchavan** [comment]
Hi,

for those who are trying to do this like me, i got it working by below defination.

```
<Griddle results={this.state.results} useExternal={true} externalSetPage={this.setPage} enableSort={false} 
        externalSetPageSize={this.setPageSize} externalMaxPage={this.state.maxPages}
        externalChangeSort={function(){}} externalSetFilter={this.setFilter}
        externalCurrentPage={this.state.page - 1 } useCustomRowComponent={true} customRowComponent={OtherComponent} customRowComponentClassName={"custom-row"}   showFilter={true} showSettings={true}/>

```

still i have one problem left about Settings where page size is changeable. it is breaking somewhere, i am looking into it.

**Comment by jitchavan** [comment]
Hi,

finally resolved  the  "Settings" page size break issue.

Examples for externalData (http://griddlegriddle.github.io/Griddle/externalData.html)
uses **Math.round** everywhere to calculate maxPages  e.g. ` maxPages: Math.round(data.count/10)` .
it should be **Math.ceil** (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil)

e.g. `maxPages: Math.round(11/10)`   will always return 1 which is wrong. should return 2.

let me know if above point is right then i will close the issue.

**Comment by ryanlanciaux** [comment]
You are right -- it should use math.ceil everywhere. If possible, would you like to make that change and submit a PR? :) 

If not we'll keep this issue open and try to address it. It should be using ceil everywhere in 1.0.

**Comment by ryanlanciaux** [comment]
This should be fixed in the recently release 1.0. Thanks and please feel free to open a new issue if any problems are encountered. 👍
